backend/dataall/modules/dataset_sharing/services/data_sharing_service.py

```diff
@@ -218,3 +200,63 @@ def revoke_share(cls, engine: Engine, share_uri: str):
             share_sm.update_state(session, share, new_share_state)
 
             return revoked_tables_succeed and revoked_folders_succeed
+
+    @staticmethod
+    def create_lf_processor(session,
+                            dataset,
+                            share,
+                            shared_tables,
+                            revoked_tables,
+                            source_environment,
+                            target_environment,
+                            env_group):
+        try:
+            catalog_details = GlueClient(database=dataset.GlueDatabaseName,
+                                         account_id=source_environment.AwsAccountId,
+                                         region=source_environment.region).get_source_catalog()
+
+            source_account_id = catalog_details.account_id if catalog_details else source_environment.AwsAccountId
```

Comment:
I'm not sure it's a good idea to do this IF here.. getting catalog details should always work. Are there valid cases where it shouldn't?

Reply:
For cases when the db is not in the catalog account , then this "if" logic will point to the source env account id